### PR TITLE
Gh 6 hdfs resource service

### PR DESCRIPTION
**Pull request by GCHQDeveloper404**

### CODE REVIEW

**Comment by GCHQDeveloper404** [comment]
ResolveParents added

File: service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+@JsonPropertyOrder(value = {"class", "conf", "dataFormat", "dataType"}, alphabetic = true)
```
Comment (ac74475) [inline service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java]: Can you rename the dataFormat to be serialisedDataFormat

File: service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+public class HDFSResourceService implements ResourceService {
```
Comment (ac74475) [inline service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java]: Can we have some Java doc explaining the purpose of this class

File: service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceDetails.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.resource.service;
+
+import uk.gov.gchq.palisade.ToStringBuilder;
+
+import java.util.regex.Pattern;
+
+public class HDFSResourceDetails {
```
Comment (ac74475) [inline service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceDetails.java]: Can we have some Java doc explaining the purpose of the class

File: service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java (hunk 1 above)
Comment (ac74475) [reply service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java]: actually ignore this one, I haven't committed that refactoring through yet.

**Pull request merged**